### PR TITLE
feat: internal support for prepared Apple Pay purchases for Stripe Billing

### PR DESCRIPTION
**Pull request by nicfix**
## Motivation / Description

Allow Stripe Billing apps to offer Apple Pay from a normal purchase button without rendering the Express Purchase Button. Prepare before the click so Safari can open the payment sheet directly from the user gesture.

## Changes introduced

- Add package-specific `prepareForQuickPurchases(params)` and opt-in `purchase({ tryWithApplePay: true })`.
- Prepare through `/checkout/start`, then complete and confirm the payment through the Stripe Billing Apple Pay flow.
- Consume prepared requests once and fall back to Stripe Checkout when preparation is missing, expired, mismatched, or Apple Pay is unavailable.
- Add a webbilling demo page for preparation, package selection, and purchase with a normal React button.
- Keep the existing Express Purchase Button and paywall configuration unchanged.

## Linear ticket (if any)

[WEB-4656](https://linear.app/revenuecat/issue/WEB-4656)

## Additional comments

Depends on the backend support in https://github.com/RevenueCat/khepri/pull/24840. The new flow is opt-in and limited to Stripe Billing apps; ordinary purchases keep using the existing checkout.

---

> [!NOTE]
> **Medium Risk**
> Touches core purchase/checkout paths and payment confirmation for Stripe Billing, though the flow is opt-in and falls back to existing checkout; depends on coordinated backend behavior.
> 
> **Overview**
> Adds **internal, opt-in Stripe Billing “quick purchase”** support so apps can open **Apple Pay from a normal button** after pre-warming the session, without using the Express Purchase Button.
> 
> **SDK:** New `prepareForQuickPurchases` plus `purchase({ tryWithApplePay: true })` cache a package-scoped preparation keyed on purchase context; prepared sessions are **single-use** and **time-limited**, with automatic **fallback to Stripe Checkout** when Apple Pay is unavailable, context mismatches, consent/discount/product-change flows apply, or the backend returns `BackendQuickPurchaseUnavailable`. Checkout start/complete now support `purchase_flow: apple_pay` and Apple Pay **billing name/address** on complete; a dedicated module drives Stripe Payment Request, tax refresh, and Payment/SetupIntent confirmation.
> 
> **Demo & tests:** Webbilling demo route/page and login shortcut; unit tests for networking, preparation edge cases, and the Apple Pay purchase helper.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 9a517ec8676218be08c8045f20e1232c35ed3a7a. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/entities/purchase-params.ts [hunk 1]
```diff
@@ -198,4 +198,34 @@ export interface PurchaseParams {
    * @internal
    */
   productChangeInfo?: ProductChangeInfo | null;
+
+  /**
+   * Tries a package-specific Apple Pay purchase prepared with
+   * {@link Purchases.prepareForQuickPurchases} before opening the normal
+   * Stripe Billing checkout.
+   *
```
Comment (nicfix) [inline src/entities/purchase-params.ts:206]: Maybe make this internal

**Comment by dalshamaa** [comment]
re-the email discussion we had offline, should we at least fail nicely here and/or in the backend?

**Review by dalshamaa — APPROVED** [review]

**Comment by jholwill** [comment]
* **#1140** <a href="https://app.graphite.com/github/pr/RevenueCat/purchases-js/1140?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1124** <a href="https://app.graphite.com/github/pr/RevenueCat/purchases-js/1124?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/RevenueCat/purchases-js/1124?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
